src/main/scala/spray/json/ProductFormats.scala

```diff
@@ -23,9 +23,18 @@ import scala.util.control.NonFatal
  * Provides the helpers for constructing custom JsonFormat implementations for types implementing the Product trait
  * (especially case classes)
  */
-trait ProductFormats extends ProductFormatsInstances {
+trait ProductFormats extends ProductFormatsInstances with ProductFormats2 {
```

Comment:
I wonder if it's a good idea to extend here? I guess it makes it easier to deprecate the old formats at some point.